pkg/api/handlers/libpod/pods.go

```diff
@@ -25,24 +31,75 @@ func PodCreate(w http.ResponseWriter, r *http.Request) {
 		runtime = r.Context().Value("runtime").(*libpod.Runtime)
 		err     error
 	)
-	var psg specgen.PodSpecGenerator
-	if err := json.NewDecoder(r.Body).Decode(&psg); err != nil {
-		utils.Error(w, "failed to decode specgen", http.StatusInternalServerError, errors.Wrap(err, "failed to decode specgen"))
+
+	opts := &entities.PodCreateOptions{Infra: true, NetFlags: &entities.NetFlags{}, Net: &entities.NetOptions{}}
+	buffer := bytes.NewBuffer(make([]byte, 0))
+	r.Body = ioutil.NopCloser(io.TeeReader(r.Body, buffer)) // create teereader so we can decode body twice, once for podOpts, once for Infra
+	if err := json.NewDecoder(r.Body).Decode(opts); err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "failed to decode specgen"))
+		return
+	}
+	flags, err := common.GenerateTempNetFlags(opts.NetFlags, opts.Infra) // a good workaround, allowing us to temprarily make netflags and use the netflags -> netOpts
```

Comment:
The API endpoint should never have to do anything with flag parsing. I do not understand the purpose of this.

Reply:
These are TEMP flags. yes they are flags but they only serve the purpose of a conduit. This function is a stepping stone between the `NetFlags` podcreateoption and actual container cli options. This is a necessity for this implementation given that many of the podcreateoptions NetOptions are of different types than what is inputted by the user AND of a different type than the container CLI options